core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -863,7 +872,36 @@ private[deploy] class Master(
     true
   }
 
-  private def decommissionWorker(worker: WorkerInfo): Unit = {
+  private def handleDecommissionWorkers(hostPorts: Seq[String]): Integer = {
+    val hostsWithoutPorts = new HashSet[String]
+    val addressToRemove = new HashSet[RpcAddress]
+    hostPorts.foreach(hp => {
```

Comment:
we can first filter the valid workers and then apply `decommissionWorker` for each worker, e.g.

`hostPorsts.filter(...).foreach(...)`

Reply:
The challenge here is that Its not a "clean filter" due to the presence of hosts without ports. The main use case is actually to decommission a given set of hosts (because the external system may not have visibility into the worker ports). Perhaps I can split this into a union of sorts ... let me see.